### ISSUE TITLE
Instructions for standing up private instance of Quetz

### ISSUE THREAD

**Issue by tpanza** [mamba-org/quetz#391]
Hello,

Would more details be available regarding how to stand up a private instance of Quetz? Two main questions...

(1) How would the "login" concept work for such an instance? I am seeing on the `README` mentions of users needing to authenticate via their `github.com` or `google.com` identities... What if neither of those are available within a private organization? What if, for example, users did have a **private** _GitLab_ account (not the public `gitlab.com`), in which login to that private GitLab system was done via username/password or SSO/SAML?

(2) Would it be possible to somehow run all/some of the conda-forge recipes and host those products on the private Quetz server? And then have users be able to `mamba install ...` by pointing their `mamba` clients to that private Quetz server?

**Comment by beenje** [comment]
Hi,
I think we have kind of the same use-case.

1. I opened a PR to add gitlab as authenticator: https://github.com/mamba-org/quetz/pull/390
It works with self-hosted GitLab instances. Note that most users don't have to login if you create only public channels. Login is only required to configure quetz (define channels) and upload new packages if you have local channels.

2. If your quetz instance has access to internet, you can configure it to proxy conda-forge channel. That would allow users to install conda-forge packages by pointing to quetz. See https://quetz.readthedocs.io/en/latest/using/mirroring.html#proxy-channels

Define conda-forge as proxy channel:

```json
{
  "name": "conda-forge",
  "description": "conda-forge mirror",
  "private": false,
  "mirror_channel_url": "https://conda.anaconda.org/conda-forge",
  "mirror_mode": "proxy"
}
```

If you deployed quetz to quetz.mydomain.org, you can set `channel_alias: https://quetz.mydomain.org/get` in your `condarc` file and continue to use the `conda-forge` channel.

**Comment by tpanza** [comment]
Thank you for this, especially for the PR #390 to add GitLab as an authenticator, and your explanation that login is not necessary for public channels.

Re (2), I was hoping to avoid _any_ involvement of `anaconda.org`. So I would rather avoid either proxying or mirroring it. How would one go about recreating conda-forge on a quetz instance simply by re-running the recipes? Is that doable / practical?

**Comment by wolfv** [comment]
@tpanza I don't think that re-running the recipes is feasible. It requires _a lot_ of CPU time, as well as multiple big machines and it's not sure that you will end up with the same binaries (as dependencies have moved etc.). It would be a herculean task (at least if we're talking about _all of conda-forge_.

there are two options:

- create your own conda-forge mirror
- use our conda-forge mirror and make a "quetz"-mirror

If you're interested in the latter, let me know and we can work something out in private (as our mirror isn't public yet :(.
